joplin/pages/base_page/models.py

```diff
@@ -48,6 +48,17 @@ class JanisBasePage(Page):
     coa_global = models.BooleanField(default=False, verbose_name='Make this a top level page')
     imported_revision_id = models.TextField(blank=True, null=True)
 
+    # Fields used for handling Publishing status
+    # pk for our publish request in Publisher dynamodb
+    publish_request_pk = models.TextField(blank=True, null=True)
+    # sk for our publish request in Publisher dynamodb
+    publish_request_sk = models.TextField(blank=True, null=True)
+    # Indicated whether a publish_request for this page been submitted to the Publisher, and we are waiting for it to finish being processed.
+    publish_request_enqueued = models.BooleanField(default=False)
+    # Has this page been published by Publisher? A "live" page may not necessarily be published to our frontend yet.
+    published = models.BooleanField(default=False, blank=True, null=True)
```

Comment:
a live page may not necessarily be published because of that liminal state of "currently being published" or heaven forbid, a page that is marked as live but b/c of error didnt actually get published?

Reply:
yup, i just read further down and answered my own question

Reply:
If there is an error with publishing and it didn't get published, then it will be in a "Publishing" status. This is by design until there's some more specific error messaging we want to add.

This is actually a good approach. Because the Publisher is smart. When we deploy the needed fix to Joplin/Janis/Publisher, the Publisher knows to sweep up all previously failed publish_requests and retry them. And our views/publish_succeeded endpoint will iterate through every page in the build and update pages that "triggered_builds" to set their status as no longer publishing.